contracts/src/internal/HyperdrivePair.sol

```diff
@@ -0,0 +1,308 @@
+    function _applyMint(
+        uint256 _maturityTime,
+        uint256 _bondAmount,
+        uint256 _governanceFee
+    ) internal {
+        // Update the amount of governance fees accrued. Since the long and
+        // short both pay the governance fee, the governance fees accrued
+        // increases by twice the governance fee.
+        _governanceFeesAccrued += 2 * _governanceFee;
+
+        // Update the average maturity time of longs and short positions and the
+        // amount of long and short positions outstanding. Everything else
+        // remains constant.
+        _marketState.longAverageMaturityTime = uint256(
+            _marketState.longAverageMaturityTime
+        )
+            .updateWeightedAverage(
+                _marketState.longsOutstanding,
+                _maturityTime * ONE, // scale up to fixed point scale
+                _bondAmount,
+                true
+            )
+            .toUint128();
+        _marketState.shortAverageMaturityTime = uint256(
+            _marketState.shortAverageMaturityTime
+        )
+            .updateWeightedAverage(
+                _marketState.shortsOutstanding,
+                _maturityTime * ONE, // scale up to fixed point scale
+                _bondAmount,
+                true
+            )
+            .toUint128();
+        _marketState.longsOutstanding += _bondAmount.toUint128();
+        _marketState.shortsOutstanding += _bondAmount.toUint128();
+    }
+
+    /// @dev Calculates the amount of bonds that can be minted and the governance
+    ///      fee from the amount of vault shares that were deposited.
+    /// @param _sharesDeposited The amount of vault shares that were deposited.
+    /// @param _vaultSharePrice The vault share price.
+    /// @param _openVaultSharePrice The vault share price at the beginning of
+    ///        the checkpoint.
+    function _calculateMint(
+        uint256 _sharesDeposited,
+        uint256 _vaultSharePrice,
+        uint256 _openVaultSharePrice
+    ) internal view returns (uint256, uint256) {
+        // In order for a certain amount of bonds to be minted, there needs to
+        // be enough base to pay the prepaid interest that has accrued since the
+        // start of the checkpoint, to pay out the face value of the bond at
+        // maturity, for the short to pay the flat fee at maturity, and for the
+        // long and short to both pay the governance fee during the mint. We can
+        // work back from this understanding to get the amount of bonds from the
+        // amount of shares deposited.
+        //
+        // sharesDeposited * vaultSharePrice = (
+        //    bondAmount + bondAmount * (c - c0) / c0 + bondAmount * flatFee +
+        //    2 * bondAmount * flatFee * governanceFee
+        // )
+        //
+        // This implies that
+        //
+        // bondAmount = shareDeposited * vaultSharePrice / (
+        //     1 + (c - c0) / c0 + flatFee + 2 * flatFee * governanceFee
```

Comment:
@jalextowle 
Informational: This part `1 + (c - c0) / c0` if we further shrink it to just `c/c0`, then in the calculation we will mitigate the potential underflow risk of `c<c0`. Although I'm not sure what's the likelihood of `c<c0` happening in reality, but I think using `c/c0` is always safer. Thoughts?